src/server.ts

```diff
@@ -1,17 +1,14 @@
-import getPort from 'get-port'
 import got, { OptionsOfUnknownResponseBody } from 'got'
 import { getContext } from './context'
 
 export async function listen () {
   const ctx = getContext()
   const { server } = ctx.options.config
-  const port = await getPort({
-    ...(server?.port && { port: Number(server?.port) })
-  })
 
-  ctx.url = 'http://localhost:' + port
+  const port = server?.port || 0
+  const { url } = await ctx.nuxt.listen(port)
```

Comment:
It would be nice if you can replace it with [listhen](https://github.com/unjs/listhen) to get url (`listen(ctx.nuxt.app)`). Since this listen function is removed in nuxt3. Also if need to normalize url, can use [ufo](https://github.com/unjs/ufo).

Reply:
Also `get` implementation should be updated with [this](https://github.com/unjs/ohmyfetch). These are standard utilities we use across nuxt repos. I didn't had a chance yet to update them.

Reply:
Thanks. This is very useful. I will fix it.

Reply:
Could you give hint how nuxt.app() should be typed?

Reply:
Here: https://github.com/nuxt/test-utils/blob/a5c7b753412c4600a2ebe4ca831f0f52f68142b1/src/context.ts#L37

we need to add it like `app: RequestListener` (`import type { RequestListener } from 'http'`)

Reply:
Got it. All works here. Fixing the rest